### PR TITLE
[CLApp] Adding capabilities to impose pre-stressing in the Serial-Parallel RoM

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
**📝 Description**
In this PR I'm adding several processes to read pre-stressing information and its loading. Several improvements to the SP-RoM have been made.

A test will be added once we reach consensus.

### CODE REVIEW

**Comment by AlejandroCornejo** [comment]
FYI @RiccardoRossi

File: applications/ConstitutiveLawsApplication/python_scripts/set_up_pre_stressed_oriented_composite_materials.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        return math.sqrt(Vector[0]**2 + Vector[1]**2+ Vector[2]**2)
+    
+    def InnerProd(self, Vector1, Vector2):
+        return Vector1[0]*Vector2[0] + Vector1[1]*Vector2[1] + Vector1[2]*Vector2[2]
+    
+    def CheckFiberParticipation(self, kf):
+        if kf <= 0.0:
+            kf = self.fiber_participation_threshold
+        elif kf > 1.0:
+            kf = 0.99999999
```
Comment (loumalouomega) [inline applications/ConstitutiveLawsApplication/python_scripts/set_up_pre_stressed_oriented_composite_materials.py:175]: Almost!

**Review by loumalouomega — APPROVED** [review]

**Comment by loumalouomega** [comment]
Looks okay

**Comment by loumalouomega** [comment]
> Looks okay

Many commits, much change, wow

image

**Pull request merged**